sdk/core/azure-core/test/ut/operation_status.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <gtest/gtest.h>
+
+#include <azure/core/operation_status.hpp>
+
+using namespace Azure::Core;
+
+TEST(OperationStatus, Basic)
+{
+  OperationStatus status = OperationStatus::Cancelled;
+  EXPECT_TRUE(status == OperationStatus::Cancelled);
+  EXPECT_TRUE(status.Get() == "Cancelled");
+  EXPECT_FALSE(status != OperationStatus::Cancelled);
+
+  status = OperationStatus::Failed;
+  EXPECT_TRUE(status == OperationStatus::Failed);
+  EXPECT_TRUE(status.Get() == "Failed");
+  EXPECT_FALSE(status != OperationStatus::Failed);
+
+  status = OperationStatus::NotStarted;
+  EXPECT_TRUE(status == OperationStatus::NotStarted);
+  EXPECT_TRUE(status.Get() == "NotStarted");
+  EXPECT_FALSE(status != OperationStatus::NotStarted);
+
+  status = OperationStatus::Running;
+  EXPECT_TRUE(status == OperationStatus::Running);
+  EXPECT_TRUE(status.Get() == "Running");
+  EXPECT_FALSE(status != OperationStatus::Running);
+
+  status = OperationStatus::Succeeded;
+  EXPECT_TRUE(status == OperationStatus::Succeeded);
+  EXPECT_TRUE(status.Get() == "Succeeded");
+  EXPECT_FALSE(status != OperationStatus::Succeeded);
+}
+
+TEST(OperationStatus, Custom)
+{
+  OperationStatus status2("CustomValue");
```

Comment:
nit: why start with 2 instead of 1?
```suggestion
  OperationStatus status1("CustomValue");
```

Reply:
Cause I had a 1 initially that didn't make sense as a test case and updating the numbering was extra work that didn't provide any value :)